CONTRIBUTING.md

```diff
@@ -2,6 +2,36 @@
 
 Contributions are very welcome. The following will provide some helpful guidelines.
 
+## How to build the project
+
+### Building locally on an OSX machine
+
+Assumptions
+
+* Java 9
+* Gradle
```

Comment:
Using the [Gradle Wrapper](https://docs.gradle.org/current/userguide/gradle_wrapper.html), which is [part of the repository](https://github.com/TNG/ArchUnit/tree/master/gradle/wrapper), should make the build independent of an external Gradle installation.

But I like the idea to state that JDK 9 is a prerequisite! :+1:

As the build should not depend on the platform (OSX vs. Linux vs. Windows), I would however remove the different OS-specific sections (and the `TBD`s as well 😉).